### PR TITLE
prepare for deduplicating a day.

### PR DESCRIPTION
**Pull request by robertbrowncc**

### CODE REVIEW

File: app/Services/StatementSearchService.php [hunk 1]
```diff
@@ -390,6 +390,25 @@ public function bulkIndexStatements(Collection $statements): void
         }
     }
 
+    public function bulkDeleteStatements(\Illuminate\Support\Collection $statements): void
```
Comment (alainvd) [inline app/Services/StatementSearchService.php:393]: maybe move to another service than the Search service ?

File: app/Console/Commands/StatementsDeDuplicateRange.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public function handle(): void
+    {
+
+        if (!Cache::get('deduplication-has-been-run', false)) {
+            Cache::forever('deduplication-has-been-run', true);
+            $chunk = $this->intifyArgument('chunk');
+            $min   = $this->argument('min') === 'default' ? DB::table('statements')->selectRaw('MIN(id) AS min')->first()->min : (int)$this->argument('min');
+            $max   = $this->argument('max') === 'default' ? DB::table('statements')->selectRaw('MAX(id) AS max')->first()->max : (int)$this->argument('max');
+
+            StatementDeDupulicateRange::dispatch($max, $min, $chunk);
```
Comment (alainvd) [inline app/Console/Commands/StatementsDeDuplicateRange.php:39]: Isn't this a typo Dedupulicate ?

File: app/Console/Commands/StatementsDeDuplicateDate.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    public function handle(DayArchiveService $day_archive_service): void
+    {
+        $chunk = $this->intifyArgument('chunk');
+        $date = $this->sanitizeDateArgument();
+
+        $min = $day_archive_service->getFirstIdOfDate($date);
+        $max = $day_archive_service->getLastIdOfDate($date);
+
+        if ($min && $max) {
+            StatementDeDupulicateRange::dispatch($max, $min, $chunk);
```
Comment (alainvd) [inline app/Console/Commands/StatementsDeDuplicateDate.php:39]: typo in the StatementDeDupulicateRange ?

File: app/Services/StatementSearchService.php (hunk 1 above)
Comment (robertbrowncc) [reply app/Services/StatementSearchService.php:393]: This is for removing statements from the search index. at present I am not doing any delete so yes i can moved to separate it better.

**Pull request closed without merging**